cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/monitor/RuntimeMonitor.java

```diff
@@ -71,6 +73,7 @@
   private final long pollTimeMillis;
   private volatile Thread runThread;
   private volatile boolean stopped;
+  private volatile boolean isRuntimeInactive;
```

Comment:
I don't think you need volatile. Also, a better name is `programFinished`, because the runtime is still active, otherwise you can't even talk to it. Moreover, boolean variable shouldn't start with `is`, because that's the name prefix for the boolean getter (e.g. a `boolean running` would have getting method `isRunning`).